From 5f251d3e56b62da6c024b28665c04c9d64f37889 Mon Sep 17 00:00:00 2001
From: Giuseppe Scrivano <gscrivan@redhat.com>
Date: Mon, 15 Dec 2014 13:33:34 +0100
Subject: [PATCH 3/3] interface: use only the name of the interface when
 defining a bridge

when defining a bridge, use only the name of a running interface
instead of including the entire XML description.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1172407

RHEL-only

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
---
 virtinst/Interface.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/virtinst/Interface.py b/virtinst/Interface.py
index 646b7e2..46f3119 100644
--- a/virtinst/Interface.py
+++ b/virtinst/Interface.py
@@ -301,9 +301,10 @@ class _InterfaceCompound(Interface):
         xml = ""
         for i in self.interfaces:
             if type(i) is str:
-                iface_xml = "    <interface name='%s' type='ethernet'/>\n" % i
+                name = i
             else:
-                iface_xml = self._indent_xml(i.XMLDesc(0), 4)
+                name = i.name()
+            iface_xml = "    <interface name='%s' type='ethernet'/>\n" % name
 
             xml += iface_xml
         return xml
-- 
2.1.0

